### PR TITLE
Adding routes, file structure, and front page 

### PR DESCRIPTION
**Pull request by lauradurandv**
Note: Front page is not yet complete
Structure: Happy to change organization
Routes: Any feedback would be great; I haven't used routes too much
General feedback is also very welcomed!
Hopefully, this can get us going.

### CODE REVIEW

**Comment by OGoncharenko** [comment]
I believe that components should be in src folder, not assets. In assets we should store images, avatars, etc.
`src/
	App.jsx
	App.css
	components/
		Teacher/
			Dashboard/
				Dashboard.jsx
				DasboardStyles.module.css
			Lesson/
				LessonItem.jsx
				LessonItemStyles.module.css
		Student/
			Dashboard/
				Dashboard.jsx
				DashboardStyles.module.css
			Profile/
				Profile.jsx
				ProfileStyles.module.css`

**Comment by lauradurandv** [comment]
I have made updates based on feedback from Olga and Frank. Frank explained that the container folder holds components that require state, for example, a form; the layout folder holds components that render on every page; and the presentational folder holds components that render data.

File: gg-pac-team3-front/src/pages/frontPage/FrontPage.jsx
Comment (ElijahMSmith) [inline gg-pac-team3-front/src/pages/frontPage/FrontPage.jsx:4]: I would expect that our NavBar should appear on every page, not just our front page. The contents of the bar we can change in the future depending on the logged in user. Could we try using an `Outlet` here for this (which would probably not go in this file, but this file can just be the actual content of the front page).

https://reactrouter.com/en/main/components/outlet

File: gg-pac-team3-front/src/pages/frontPage/Panel.jsx
Comment (ElijahMSmith) [inline gg-pac-team3-front/src/pages/frontPage/Panel.jsx:6]: We don't need a fragment here if we keep everything in a `<div>` anyways.

**Comment by ElijahMSmith** [comment]
I see a `package-lock.json` at the top-level of your branch. I think this was removed in a previous commit, so make sure to `git rebase master` on this branch to pull that commit into your branch.

File: gg-pac-team3-front/src/App.jsx
Comment (ElijahMSmith) [inline gg-pac-team3-front/src/App.jsx:11]: This might work to do `FrontPage()` since it's a functional component, I honestly don't know. For the sake of clarity that this is a component, if not correctness, we should do `<FrontPage/>` inside this element attribute instead.

File: gg-pac-team3-front/src/App.jsx
Comment (ElijahMSmith) [inline gg-pac-team3-front/src/App.jsx:2]: This import does not work for me. When I run the project, I get an error in the browser that it cannot find this import. When I change it to `./pages/frontPage/FrontPage` it works, which is the name of the component rather than the name of the file.

File: gg-pac-team3-front/src/pages/frontPage/FrontPage.jsx
Comment (ElijahMSmith) [inline gg-pac-team3-front/src/pages/frontPage/FrontPage.jsx:1]: Similarly, `navBlue` here should be `NavBlue`, the name of our component, for this import to work.

**Review by ElijahMSmith — CHANGES_REQUESTED** [review]
Does this code compile and run on your machine? If not, we should be extra careful to make sure code works before opening it for review. If it does work fine, let's try to figure out what the reason might be so we can all have the same setup.

File: gg-pac-team3-front/src/App.jsx
Comment (lauradurandv) [reply gg-pac-team3-front/src/App.jsx:2]: Got you! sorry lots of file changes were made last night.

File: gg-pac-team3-front/src/App.jsx
Comment (lauradurandv) [reply gg-pac-team3-front/src/App.jsx:11]: Got it!

File: gg-pac-team3-front/src/components/layout/Nav.jsx
Comment (mattschmitz) [inline gg-pac-team3-front/src/components/layout/Nav.jsx:14]: I don't see a reason to store navLinks in State, since they are entirely determined by `isLoggedIn` and do not change except when that prop changes. If we add more complex branching logic down the line this could change, but for now I would say lets not add unnecessary state management. Also, since this is ordered data, I would just store in an array. This might look something like:

```
const navLinks = loggedIn
    ? ['Dashboard', 'Account', 'Logout']
    : ['About', 'Login', 'Register'];
  ```
and in the return map over these like

```
          <ul>
            {navLinksNew.map((link) => (
              <li key={link}>
                <NavLink to={link.toLowerCase()}>{link}</NavLink>
              </li>
            ))}
          </ul>
```

(it's typical for routes to be lowercase just fyi)

File: gg-pac-team3-front/src/components/layout/Nav.jsx
Comment (mattschmitz) [inline gg-pac-team3-front/src/components/layout/Nav.jsx:6]: Use arrow function syntax when possible, i.e.

`const Nav = (props) => {`

`const` functions have a couple of benefits over old school function declarations, mainly that they are immutable and avoid wierdness with the `this` keyword.

**Review by mattschmitz — CHANGES_REQUESTED** [review]
This looks really good! Left a few comments, all minor. I like the size of this PR - it's not trying to do too many things at once, which makes review harder.

File: gg-pac-team3-front/src/App.css
Comment (lauradurandv) [reply gg-pac-team3-front/src/App.css:8]: will remove!

File: gg-pac-team3-front/src/index.css
Comment (lauradurandv) [reply gg-pac-team3-front/src/index.css:30]: Very helpful. I always wondered about resetting the css completely but was unsure about the standard.

File: gg-pac-team3-front/src/components/layout/Nav.jsx
Comment (lauradurandv) [reply gg-pac-team3-front/src/components/layout/Nav.jsx:14]: Makes sense. I wanted to try something like map earlier but because I had it in an obj it would not let me do a for  or if else statement in return

File: gg-pac-team3-front/src/pages/frontPage/Panel.jsx
Comment (ElijahMSmith) [inline gg-pac-team3-front/src/pages/frontPage/Panel.jsx:5]: Now would be a good time to install a formatter like `Prettier` and turn on `format on save` and `format on paste` in your VS Code settings.

File: gg-pac-team3-front/src/components/layout/Nav.jsx
Comment (ElijahMSmith) [inline gg-pac-team3-front/src/components/layout/Nav.jsx:28]: I don't think this will work. In this example (https://reactrouter.com/en/main/components/outlet), the outlet is placed in a component that is a parent route to other components/pages. Thus, when we go to one of this child paths, it shows the `Dashboard` component and fits in the component assigned to the specific child route in the place of the `Outlet`.

Here, I think we should have a route structure similar to what the example has, like:

```
function App() {
  ....
  return (
    <Routes>
      <Route path="/" element={<Panel/>}>
        <Route
          path="frontPage"
          element={<FrontPage/>}
        />
        // Other routes in the future go here
      </Route>
    </Routes>
  );
}
```

Then our panel page should be like:
```
return (
    <Nav/>
    <Outlet/>
);
```

and the front page can stay as it currently is. This way, `Panel` just defines the Navbar -> Page body structure and whatever page the user navigates to can be slotted into the Outlet.

**Review by ElijahMSmith — APPROVED** [review]
Make sure to run a formatter on these pages before you merge it! There are lots of extra lines and messy indentations. I am approving now so that you can make these few changes and then merge without a re-review.

**Comment by ElijahMSmith** [comment]
Also, if there is a button down at the bottom when you are ready to merge to "squash" these 4 commits into 1, please do so! Ideally, each commit is one complete unit of change that takes our project from one working state to another.

File: gg-pac-team3-front/src/components/layout/Nav.jsx
Comment (lauradurandv) [reply gg-pac-team3-front/src/components/layout/Nav.jsx]: Switched to arrow functions!

File: gg-pac-team3-front/src/components/layout/Nav.jsx
Comment (mattschmitz) [inline gg-pac-team3-front/src/components/layout/Nav.jsx:26]: please fix the indentation

File: gg-pac-team3-front/src/components/layout/Nav.jsx
Comment (mattschmitz) [reply gg-pac-team3-front/src/components/layout/Nav.jsx:28]: @lauradurandv did you figure this out?

**Review by mattschmitz — CHANGES_REQUESTED** [review]
Looks good, just need to clean up the formatting of the JSX. Prettier will do this automatically, or you can do manually.

File: gg-pac-team3-front/src/components/layout/Nav.jsx
Comment (lauradurandv) [reply gg-pac-team3-front/src/components/layout/Nav.jsx:28]: I believe I figured it out.

File: gg-pac-team3-front/src/pages/frontPage/Panel.jsx
Comment (lauradurandv) [reply gg-pac-team3-front/src/pages/frontPage/Panel.jsx:6]: decided to manually fix it.

File: gg-pac-team3-front/src/components/layout/Nav.jsx
Comment (lauradurandv) [reply gg-pac-team3-front/src/components/layout/Nav.jsx]: Fixed!

**Review by mattschmitz — APPROVED** [review]
Nice work!

**Comment by lauradurandv** [comment]
Rebasing and merging: routes, file structure, and front page.

**Pull request merged**
